src/infra/core/src/ingest/fetch_service/evm.rs

```diff
@@ -0,0 +1,300 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::HashMap;
+use std::path::Path;
+use std::sync::Arc;
+
+use futures::TryStreamExt;
+use internal_error::{ErrorIntoInternal, ResultIntoInternal};
+use kamu_core::*;
+use kamu_datasets::DatasetEnvVar;
+use opendatafabric::*;
+
+use super::*;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+impl FetchService {
+    // TODO: FIXME: This implementation is overly complex due to DataFusion's poor
+    // support of streaming / unbounded sources.
+    //
+    // In future datafusion-ethers should implement queries as unbounded source and
+    // `DataFrame::execute_stream()` should not only yield data batches, but also
+    // provide access to the state of the underlying stream for us to know how
+    // far in the block range we have scanned. Since currently we can't access
+    // the state of streaming - we can't interrupt the stream to resume later.
+    // The implementation therefore uses DataFusion only to analyze SQL and
+    // convert the WHERE clause into a filter, and then calls ETH RPC directly
+    // to scan through block ranges.
+    //
+    // TODO: Account for re-orgs
+    #[cfg(feature = "ingest-evm")]
```

Comment:
We can remove this conditional compilation since the module itself is hidden behind the feature gate (in `mod.rs`)

Reply:
yep, that was my original intention but I guess I got distracted